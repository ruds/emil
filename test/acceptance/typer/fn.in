val tail = fn nil => nil | ((::) (_, t)) => t;
fn (a, b) => (a, tail b);
fn (a, b) : ('a * 'b list) => (a, tail b);
fn (a, b) : ('a * 'a list) => (a, tail b);
fn (a, b) : ('a * 'b) => (a, tail b);
fn (a, 2) => a
 | (_, 3) => 0
 | (a, _) => a;
fn (a, 2) => a
 | (_, 3) => 0;
fn (a, 2) => a
 | (_, 3) => 0
 | (a, _) => a
 | _ => 12;
 fn {a, b=5, ...} => a | {b=7, c, ...} => c | {a, b, c} => a;
 fn {a, b=5, ...} => a | {b=7, c, ...} => c | {a, ...} => a;
 fn {a, b=5, ...} : {a: 'a, b: bigint, c: 'a} => a
  | {b=7, c, ...} => c
  | {a, ...} => a;
