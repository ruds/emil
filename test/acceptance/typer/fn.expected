@0001:
val tail : 'a list -> 'a list
@0002:
val it : ('a * 'b list) -> ('a * 'b list)
@0003:
val it : ('a * 'b list) -> ('a * 'b list)
@0004:
val it : ('a * 'a list) -> ('a * 'a list)
@0005: ERROR
@0006:
val it : (bigint * bigint) -> bigint
@0009: WARNING
@0009:
val it : (bigint * bigint) -> bigint
@0011: ERROR
@0015:
val it : {a: 'a, b: bigint, c: 'a} -> 'a
@0016: ERROR
@0017:
val it : {a: 'a, b: bigint, c: 'a} -> 'a
@0020:
